src/yield/library.cairo

```diff
@@ -420,32 +392,33 @@ namespace CarbonableYielder_assert {
         // [Check] Snapshot timestamp is lower than the current time
         let (current_time) = get_block_timestamp();
         let (snapshoted_time) = CarbonableYielder.snapshoted_time();
-        let is_lower = is_le(current_time, snapshoted_time);
         with_attr error_message(
                 "CarbonableYielder: cannot snapshot at a sooner time that previous snapshot") {
-            assert is_lower = FALSE;
+            assert_in_range(snapshoted_time, 0, current_time);
         }
 
         // [Check] Project is setup
         let (carbonable_project_address) = CarbonableOffseter.carbonable_project_address();
-        let (setup) = ICarbonableProject.isSetup(contract_address=carbonable_project_address);
+        let (slot) = CarbonableOffseter.carbonable_project_slot();
+        let (setup) = ICarbonableProject.isSetup(
+            contract_address=carbonable_project_address, slot=slot
+        );
         with_attr error_message("CarbonableYielder: cannot snapshot if the project is not set up") {
             assert setup = TRUE;
         }
 
         // [Check] At least 1 user registered
-        let (users_len, users) = CarbonableOffseter.registered_users();
-        let not_zero = is_not_zero(users_len);
+        let (count) = CarbonableOffseter.total_user_count();
         with_attr error_message(
                 "CarbonableYielder: cannot snapshot or create vestings if no user has registered") {
-            assert not_zero = TRUE;
+            assert_not_zero(count);
         }
-        return (users_len=users_len, users=users);
+        return ();
     }
 
     func is_vestable{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
```

Comment:
Same here, `vestable`.
I would prefer that `is_something` returns a boolean without failing.